### PR TITLE
[SPARK-19227][SPARK-19251] remove unused imports and outdated comments

### PR DESCRIPTION
**Pull request by uncleGen**
## What changes were proposed in this pull request?
remove ununsed imports and outdated comments, and fix some minor code style issue.

## How was this patch tested?
existing ut

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
I know this is acceptable assuming from the history. However, I have seen a lot of unused imports across the code base. I think it'd be nicer if the same instances are checked at least within the same package.

**Comment by uncleGen** [comment]
This work does not change any code, but just delete unused imports and fix some code style issue. 
cc @srowen

**Comment by uncleGen** [comment]
retest this please

**Comment by f7753** [comment]
Nice work

File: core/src/main/java/org/apache/spark/api/java/JavaFutureAction.java [hunk 1]
```diff
@@ -17,7 +17,6 @@
 
 package org.apache.spark.api.java;
 
-
```
Comment (srowen) [inline core/src/main/java/org/apache/spark/api/java/JavaFutureAction.java]: This type of change shouldn't be made. Removing dead imports we usually don't bother with but maybe one sweep of all the code is OK.

**Review by srowen — CHANGES_REQUESTED** [review]
This seems to be tagged to the wrong JIRA

File: core/src/main/scala/org/apache/spark/executor/OutputMetrics.scala [hunk 2]
```diff
@@ -20,7 +20,6 @@ package org.apache.spark.executor
 import org.apache.spark.annotation.DeveloperApi
 import org.apache.spark.util.LongAccumulator
 
-
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/executor/OutputMetrics.scala]: why are you removing the line here? having two blank lines to separate top level classes is actually a style.

File: core/src/main/java/org/apache/spark/api/java/JavaFutureAction.java (hunk 1 above)
Comment (uncleGen) [inline core/src/main/java/org/apache/spark/api/java/JavaFutureAction.java]: Get it

**Comment by uncleGen** [comment]
retest this please.

**Comment by uncleGen** [comment]
No errors, but process was terminated by signal 9.

retest this please.

**Comment by uncleGen** [comment]
retest this please.

File: core/src/main/scala/org/apache/spark/rpc/netty/NettyStreamManager.scala [hunk 3]
```diff
@@ -19,6 +19,7 @@ package org.apache.spark.rpc.netty
 import java.io.File
 import java.util.concurrent.ConcurrentHashMap
 
+import org.apache.spark.SparkContext
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/rpc/netty/NettyStreamManager.scala]: Was this intentional? I would have guessed this wouldn't compile without this import before if it were needed.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala [hunk 4]
```diff
@@ -29,7 +29,6 @@ import org.apache.spark.sql.catalyst.util.quoteIdentifier
 import org.apache.spark.sql.types.StructType
 
 
-
```
Comment (srowen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala]: Back this out too if you please to be consistent

**Review by srowen — COMMENTED** [review]
OK, I know we've said don't bother with import fixups before, but I do think the occasional complete sweep is worthwhile, in a minor release. 

I had a few comments but I'm OK with merging this for master otherwise.

**Comment by srowen** [comment]
@uncleGen Oh, I see now that SPARK-19227 is just a doc issue and you fixed it here. That also isn't something that needs a JIRA. But if you resolve it here, also add `[SPARK-19227]` to the title as well.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**